sdk/go/common/resource/plugin/rpc.go

```diff
@@ -148,12 +149,35 @@ func MarshalPropertyValue(v resource.PropertyValue, opts MarshalOptions) (*struc
 		}
 		return nil, nil // return nil and the caller will ignore it.
 	} else if v.IsOutput() {
-		// Note that at the moment we don't differentiate between computed and output properties on the wire.  As
-		// a result, they will show up as computed on the other end.  This distinction isn't currently interesting.
-		if opts.KeepUnknowns {
-			return marshalUnknownProperty(v.OutputValue().Element, opts), nil
+		if !opts.KeepOutputValues {
+			if !v.OutputValue().Known {
+				// Unknown outputs are marshaled the same as Computed.
+				computed := resource.MakeComputed(v.OutputValue().Element)
+				return MarshalPropertyValue(computed, opts)
+			} else if v.OutputValue().Secret {
+				secret := resource.MakeSecret(v.OutputValue().Element)
+				return MarshalPropertyValue(secret, opts)
+			}
+			return MarshalPropertyValue(v.OutputValue().Element, opts)
```

Comment:
How do we marshal secret unknowns today? Do those even come up?

Reply:
I didn't think they came up in practice -- the [Node.js](https://github.com/pulumi/pulumi/blob/e696fb6c5017b0ebffcb56069438b01dfa5ab089/sdk/nodejs/runtime/rpc.ts#L345-L354), [Python](https://github.com/pulumi/pulumi/blob/e696fb6c5017b0ebffcb56069438b01dfa5ab089/sdk/python/lib/pulumi/runtime/rpc.py#L264-L272), and [.NET](https://github.com/pulumi/pulumi/blob/e696fb6c5017b0ebffcb56069438b01dfa5ab089/sdk/dotnet/Pulumi/Serialization/Serializer.cs#L147-L157) SDKs will marshal an unknown secret as just an unknown value without the secret information. However, the [Go](https://github.com/pulumi/pulumi/blob/e696fb6c5017b0ebffcb56069438b01dfa5ab089/sdk/go/pulumi/rpc.go#L223-L225) SDK will keep it as an unknown value wrapped as a secret, so it is possible. I updated the code here to do the same.